build/Common.prod.props

```diff
@@ -74,7 +74,7 @@
   </PropertyGroup>
 
   <!--PublicApi Analyzer-->
-  <ItemGroup>
+  <ItemGroup Condition="!($(ExcludeFromPublicApiAnalyzer) == 'true')">
```

Comment:
I thought there was already a way to accomplish this, but `RunApiCompat=false` did not accomplish what I needed. It seems `RunApiCompat` still requires the presence of the publicApi files.

The problem is, the first build generates the code then fails because then the generated API is not declared in the public API. If I add things to the publicApi files the build works, but then if I do a clean build - wiping out the generated code - the public API analyzer runs prior to generation and fails because there are items declared in the public API that are not in the code (because it hasn't been generated yet).

Reply:
Is this project needed, or the entire code-gen can be done as a pre step in the OTLP exporter project?

E.g.

1. OTLP exporter project build started
2. fetch the proto files based on the version specified in the OTLP exporter project
3. run command to build the generated code (it is possible that we need to generate a temp project here) - since this is not tight to the overall project, many rules won't apply (e.g. style check, api compat, etc.)
4. copy the output proto binary (DLLs, XML docs and PDF files) to the target place
5. build the OTLP exporter
6. pack the outputs if nuget pack step is enabled

Reply:
Stepping back question - do we in general prefer keeping generated code in the repo or not?

I guess folks would have very different opinions. I've seen 80% projects keep generated code in the repo while 20% try to be mean-and-lean (and sometimes when issue happened, folks had hard time debugging stuff due to the lack of source/symbol).

Reply:
We do not currently keep generated code in the repo. We do keep a copy of the proto files in the OTLP exporter project that it generates the code from, but it internalizes it so public API is not an issue.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/2c9501fcd54aa52f18b1b98e90ce9004a890a89b/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OpenTelemetry.Exporter.OpenTelemetryProtocol.csproj#L54-L58

This PR demonstrates we don't necessarily even need to keep a copy of the protos in the repo. Technically, we could take this same approach on the OTLP exporter project itself.

Reply:
Yeah, Jaeger exporter took a different approach here https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/Protocol.

Reply:
I think the value of publishing the .NET generated proto code as a nuget package is a question that can be considered separately.

I find that it is somewhat useful. I have worked on some internal tooling requiring me to interact directly with the protos. It's not hugely difficult to acquire the protos and generate the code, but it is far easier to just reference a nuget package.

There is a precedent for this from some of the other language SIGs
* https://github.com/open-telemetry/opentelemetry-proto-java
* https://github.com/open-telemetry/opentelemetry-proto-go

They have an entire separate repo. I suppose we could consider this as well.

Reply:
Yep, IIRC in Python we were also having a separate repo (so the change history is not entangled with the main project), and things can be rolled back/forward independently.

Reply:
> Yeah, Jaeger exporter took a different approach here https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/Protocol.

Ah, yes, in talking with @CodeBlanch he pointed this to me as well. I think maintaining the generated thrift code for jaeger probably makes sense here. I'm not familiar with the tooling for generating this, so not sure how it differs from the MSBuild tooling for generating code from protos.

Reply:
> Yep, IIRC in Python we were also having a separate repo (so the change history is not entangled with the main project), and things can be rolled back/forward independently.

It looks like Python maintains this in their core repo

Reply:
I meant this:

* https://github.com/census-instrumentation/opencensus-proto/tree/master/gen-python
* https://github.com/census-instrumentation/opencensus-python/tree/master/contrib/opencensus-ext-ocagent/opencensus/ext/ocagent